composer/utils/libcloud_object_store.py

```diff
@@ -183,29 +184,47 @@ def get_object_size(self, object_name: str) -> int:
         """
         return self._get_object(object_name).size
 
-    def download_object(self,
-                        object_name: str,
-                        destination_path: str,
-                        overwrite_existing: bool = False,
-                        delete_on_failure: bool = True):
+    def download_object(
+        self,
+        object_name: str,
+        destination_path: str,
+        overwrite_existing: bool = False,
+    ):
         """Download an object to the specified destination path.
 
         .. seealso:: :meth:`libcloud.storage.base.StorageDriver.download_object`.
 
         Args:
             object_name (str): The name of the object to download.
-
             destination_path (str): Full path to a file or a directory where the incoming file will be saved.
-
             overwrite_existing (bool, optional): Set to ``True`` to overwrite an existing file. (default: ``False``)
-            delete_on_failure (bool, optional): Set to ``True`` to delete a partially downloaded file if
-                the download was not successful (hash mismatch / file size). (default: ``True``)
         """
+        if os.path.exists(destination_path) and not overwrite_existing:
+            # If the file already exits, short-circuit and skip the download
+            raise FileExistsError(
+                f"destination_path {destination_path} exists and overwrite_existing was set to False.")
+
         obj = self._get_object(object_name)
-        self._provider.download_object(obj=obj,
-                                       destination_path=destination_path,
-                                       overwrite_existing=overwrite_existing,
-                                       delete_on_failure=delete_on_failure)
+        # Download first to a tempfile, and then rename, in case if the file gets corrupted in transit
+        tmp_filepath = destination_path + f".{time.time()}.tmp"
```

Comment:
I would prefer generating random file names with `uuid.uuid4()` -- I believe certain systems only support second level granularity with `time.time()`, so this could (while extremely rare) have namespace collisions.